repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/InspectPreviousBuild.java

```diff
@@ -0,0 +1,61 @@
+package fr.inria.spirals.repairnator.process.step;
+
+import fr.inria.spirals.jtravis.entities.Build;
+import fr.inria.spirals.jtravis.entities.BuildStatus;
+import fr.inria.spirals.jtravis.helpers.BuildHelper;
+import fr.inria.spirals.repairnator.process.inspectors.ProjectInspector;
+import fr.inria.spirals.repairnator.process.inspectors.ProjectInspector4Bears;
+
+/**
+ * Created by fermadeiral.
+ */
+public class InspectPreviousBuild extends AbstractStep {
+
+	public InspectPreviousBuild(ProjectInspector inspector) {
+		super(inspector);
+	}
+
+	@Override
+	protected void businessExecute() {
+		if (this.inspector instanceof ProjectInspector4Bears) {
+			
+			this.getLogger().debug("Inspecting previous build...");
+			
+			((ProjectInspector4Bears)this.inspector).setPreviousBuildFlag(true);
+			
+			Build previousBuild = BuildHelper.getLastBuildOfSameBranchOfStatusBeforeBuild(this.inspector.getBuild(), null);
+			
+			if (previousBuild != null) {
+				((ProjectInspector4Bears)this.inspector).setPreviousBuild(previousBuild);
+				
+				this.getLogger().debug("Build: " + this.inspector.getBuild().getId());
+				this.getLogger().debug("Previous build: " + previousBuild.getId());
+				
+				if (previousBuild.getBuildStatus() == BuildStatus.FAILED) {
+					AbstractStep testPreviousFailingBuild = new TestPreviousFailingBuild(this.inspector, previousBuild);
```

Comment:
Ok these steps should be already added in the inspector, and you should use the "shouldStop" flag to stop depending the status. Then the pipeline is already built and you don't need to do all this stuff here.